### ISSUE TITLE
AssertionError null

### ISSUE THREAD

**Issue by himamis** [google/j2objc#1640]
When translating this file
```
package com.company;

public class Main {

    enum AnEnum {
        A, B
    }

    public static void main(String[] args) {
        AnEnum anEnum = AnEnum.A;
        boolean aBool = true;
        if (anEnum == AnEnum.B == aBool) {
            System.out.println("Equals");
        } else {
            System.out.println("Not equals");
        }
    }
}
```
I get
```
error: internal error translating "/Users/***/IdeaProjects/bugxemple/src/com/company/Main.java"
com.google.devtools.j2objc.ast.TreeVisitorError: /Users/***/IdeaProjects/bugxemple/src/com/company/Main.java:12: AssertionError: null
    at com.google.devtools.j2objc.translate.OperatorRewriter.getInfixFunction(OperatorRewriter.java:428)
    at com.google.devtools.j2objc.translate.OperatorRewriter.endVisit(OperatorRewriter.java:112)
    at com.google.devtools.j2objc.ast.InfixExpression.acceptInner(InfixExpression.java:148)
    at com.google.devtools.j2objc.ast.TreeNode.accept(TreeNode.java:95)
    at com.google.devtools.j2objc.ast.ChildLink.accept(ChildLink.java:80)
    at com.google.devtools.j2objc.ast.IfStatement.acceptInner(IfStatement.java:70)
    at com.google.devtools.j2objc.ast.TreeNode.accept(TreeNode.java:95)
    at com.google.devtools.j2objc.ast.ChildLink.accept(ChildLink.java:80)
    at com.google.devtools.j2objc.ast.ChildList.accept(ChildList.java:90)
    at com.google.devtools.j2objc.ast.Block.acceptInner(Block.java:55)
    at com.google.devtools.j2objc.ast.TreeNode.accept(TreeNode.java:95)
    at com.google.devtools.j2objc.ast.ChildLink.accept(ChildLink.java:80)
    at com.google.devtools.j2objc.ast.FunctionDeclaration.acceptInner(FunctionDeclaration.java:100)
    at com.google.devtools.j2objc.ast.TreeNode.accept(TreeNode.java:95)
    at com.google.devtools.j2objc.ast.ChildLink.accept(ChildLink.java:80)
    at com.google.devtools.j2objc.ast.ChildList.accept(ChildList.java:90)
    at com.google.devtools.j2objc.ast.TypeDeclaration.acceptInner(TypeDeclaration.java:95)
    at com.google.devtools.j2objc.ast.TreeNode.accept(TreeNode.java:95)
    at com.google.devtools.j2objc.ast.ChildLink.accept(ChildLink.java:80)
    at com.google.devtools.j2objc.ast.ChildList.accept(ChildList.java:90)
    at com.google.devtools.j2objc.ast.CompilationUnit.acceptInner(CompilationUnit.java:181)
    at com.google.devtools.j2objc.ast.TreeNode.accept(TreeNode.java:95)
    at com.google.devtools.j2objc.ast.UnitTreeVisitor.run(UnitTreeVisitor.java:50)
    at com.google.devtools.j2objc.pipeline.TranslationProcessor.applyMutations(TranslationProcessor.java:286)
    at com.google.devtools.j2objc.pipeline.TranslationProcessor.processConvertedTree(TranslationProcessor.java:107)
    at com.google.devtools.j2objc.pipeline.FileProcessor.processCompiledSource(FileProcessor.java:154)
    at com.google.devtools.j2objc.pipeline.FileProcessor$1.handleParsedUnit(FileProcessor.java:132)
    at com.google.devtools.j2objc.javac.JavacParser.parseFiles(JavacParser.java:205)
    at com.google.devtools.j2objc.pipeline.FileProcessor.processBatch(FileProcessor.java:137)
    at com.google.devtools.j2objc.pipeline.FileProcessor.processInputs(FileProcessor.java:67)
    at com.google.devtools.j2objc.pipeline.TranslationProcessor.processInputs(TranslationProcessor.java:83)
    at com.google.devtools.j2objc.J2ObjC.run(J2ObjC.java:127)
    at com.google.devtools.j2objc.J2ObjC.main(J2ObjC.java:176)
```

Workaround is to rewrite the source to

```
package com.company;

public class Main {

    enum AnEnum {
        A, B
    }

    public static void main(String[] args) {
        AnEnum anEnum = AnEnum.A;
        boolean aBool = true;
        boolean anotherBool = anEnum == AnEnum.B;
        if (anotherBool == aBool) {
            System.out.println("Equals");
        } else {
            System.out.println("Not equals");
        }
    }
}
```

**Comment by himamis** [comment]
I think it's related to #1480

**Comment by tomball** [comment]
Yes, this works with the current source, thanks to the fix to #1480 (https://github.com/google/j2objc/commit/c960e45598ba98d7d754f3e51a5d1b8a4ab82b77). If you are using a pre-built version of j2objc, consider instead building from the latest source.
